File: Desktop/media-tracker-desktop/media-tracker-desktop/Models/Steam/Steam_Model.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace media_tracker_desktop.Models.Steam
{
    // This model represents each game object in the games array in the response json object returned by the GetOwnedGames endpoint of the SteamAPI.
    public class Steam_Model
    {
        [JsonProperty("appid")]
        public int AppID { get; }


        [JsonProperty("name")]
        public string Name { get; } = string.Empty;


        [JsonProperty("playtime_forever")]
        public int PlaytimeForever { get; }


        [JsonProperty("img_icon_url")]
        public string ImgIconUrl { get; }


        [JsonProperty("has_community_visible_stats")]
        public bool HasCommunityStats { get; }


        [JsonProperty("playtime_windows_forever")]
        public int PlaytimeWindows { get; }


        [JsonProperty("playtime_mac_forever")]
        public int PlaytimeMac { get; }


        [JsonProperty("playtime_linux_forever")]
        public int PlaytimeLinux { get; }


        [JsonProperty("playtime_deck_forever")]
        public int PlaytimeDeck { get; }


        [JsonProperty("rtime_last_played")]
        public int RTimeLastPlayed { get; }


        [JsonProperty("playtime_disconnected")]
        public int PlaytimeDisconnected { get; }


        [JsonConstructor]
        public Steam_Model(int appId, string name, int playtimeForever, string imgIconUrl, bool hasCommunityStats, int playtimeWindows, int playtimeMac, int playtimeLinux, int playtimeDeck, int rtimeLastPlayed, int playtimeDisconnected)
        {
            this.AppID = appId;
            this.Name = name;
            this.PlaytimeForever = playtimeForever;
            this.ImgIconUrl = imgIconUrl;
            this.HasCommunityStats = hasCommunityStats;
            this.PlaytimeWindows = playtimeWindows;
            this.PlaytimeMac = playtimeMac;
            this.PlaytimeLinux = playtimeLinux;
            this.PlaytimeDeck = playtimeDeck;
            this.RTimeLastPlayed = rtimeLastPlayed;
            this.PlaytimeDisconnected = playtimeDisconnected;
        }
    }
}

File: Mobile/media_tracker_mobile/lib/models/steam/steam_model.dart
class SteamGame {
  final int appId;
  final String name;
  final int playtimeForever;
  final String imgIconUrl;
  final bool hasCommunityStats;
  final int playtimeWindows;
  final int playtimeMac;
  final int playtimeLinux;
  final int playtimeDeck;
  final int rtimeLastPlayed;
  final int playtimeDisconnected;
  bool isFavorite;

  SteamGame({
    required this.appId,
    required this.name,
    required this.playtimeForever,
    required this.imgIconUrl,
    required this.hasCommunityStats,
    required this.playtimeWindows,
    required this.playtimeMac,
    required this.playtimeLinux,
    required this.playtimeDeck,
    required this.rtimeLastPlayed,
    required this.playtimeDisconnected,
    this.isFavorite = false,
  });

  factory SteamGame.fromJson(Map<String, dynamic> json) {
    return SteamGame(
      appId: json['appid'],
      name: json['name'],
      playtimeForever: json['playtime_forever'],
      imgIconUrl: json['img_icon_url'],
      hasCommunityStats: json['has_community_visible_stats'] ?? false,
      playtimeWindows: json['playtime_windows_forever'] ?? 0,
      playtimeMac: json['playtime_mac_forever'] ?? 0,
      playtimeLinux: json['playtime_linux_forever'] ?? 0,
      playtimeDeck: json['playtime_deck_forever'] ?? 0,
      rtimeLastPlayed: json['rtime_last_played'] ?? 0,
      playtimeDisconnected: json['playtime_disconnected'] ?? 0,
      isFavorite: false,
    );
  }

  SteamGame copyWith({
    int? appId,
    String? name,
    int? playtimeForever,
    String? imgIconUrl,
    bool? hasCommunityStats,
    int? playtimeWindows,
    int? playtimeMac,
    int? playtimeLinux,
    int? playtimeDeck,
    int? rtimeLastPlayed,
    int? playtimeDisconnected,
    bool? isFavorite,
  }) {
    return SteamGame(
      appId: appId ?? this.appId,
      name: name ?? this.name,
      playtimeForever: playtimeForever ?? this.playtimeForever,
      imgIconUrl: imgIconUrl ?? this.imgIconUrl,
      hasCommunityStats: hasCommunityStats ?? this.hasCommunityStats,
      playtimeWindows: playtimeWindows ?? this.playtimeWindows,
      playtimeMac: playtimeMac ?? this.playtimeMac,
      playtimeLinux: playtimeLinux ?? this.playtimeLinux,
      playtimeDeck: playtimeDeck ?? this.playtimeDeck,
      rtimeLastPlayed: rtimeLastPlayed ?? this.rtimeLastPlayed,
      playtimeDisconnected: playtimeDisconnected ?? this.playtimeDisconnected,
      isFavorite: isFavorite ?? this.isFavorite,
    );
  }
}
